events/conn_test.go

```diff
@@ -0,0 +1,40 @@
+package events
+
+import (
+	"context"
+	"github.com/stretchr/testify/assert"
+	"sync"
+	"testing"
+	"time"
+)
+
+func TestNATSConnectionPool_Acquire(t *testing.T) {
+	manager := NewManager().(*manager)
+
+	pool := manager.Pool()
+
+	err := manager.Start()
+	assert.NoError(t, err)
+
+	defer manager.Shutdown()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	wc := sync.WaitGroup{}
+	wc.Add(500)
```

Comment:
when 10 just doesn't cut it?

Reply:
I wanted to catch possible race conditions quickly but 500 is a bit too extreme 😅 Changed it to 50 which should be fine